src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java

```diff
@@ -45,37 +74,56 @@ private Network getNetwork(UUID networkUuid) {
         LoadFlowParameters params = parameters != null ? parameters : new LoadFlowParameters();
         LoadFlowResult result;
 
         LoadFlow.Runner runner = LoadFlow.find(provider != null ? provider : DEFAULT_PROVIDER);
+        Reporter reporter;
         if (otherNetworksUuid.isEmpty()) {
             Network network = getNetwork(networkUuid);
 
+            reporter = new ReporterModel("loadFlow", "loadFlow");
             // launch the load flow on the network
-            result = runner.run(network, params);
+            result = runner.run(network, network.getVariantManager().getWorkingVariantId(), LocalComputationManager.getDefault(), params, reporter);
             // flush network in the network store
             if (result.isOk()) {
                 networkStoreService.flush(network);
             }
-
+            sendReport(networkUuid, reporter, true);
         } else {
+            boolean doReport = reportId.isPresent() && reportName.isPresent();
             // creation of the merging view and merging the networks
             MergingView mergingView = MergingView.create("merged", "iidm");
             List<Network> networks = new ArrayList<>();
             networks.add(getNetwork(networkUuid));
             otherNetworksUuid.forEach(uuid -> networks.add(getNetwork(uuid)));
-            mergingView.merge(networks.toArray(new Network[networks.size()]));
+            mergingView.merge(networks.toArray(new Network[0]));
 
             // launch the load flow on the merging view
-            result = runner.run(mergingView, params);
+            reporter = doReport ? new ReporterModel(reportName.get(), reportName.get()) : Reporter.NO_OP;
+            result = runner.run(mergingView, mergingView.getVariantManager().getWorkingVariantId(), LocalComputationManager.getDefault(), params, reporter);
             if (result.isOk()) {
                 // flush each network of the merging view in the network store
                 networks.forEach(network -> networkStoreService.flush(network));
             }
+            if (doReport) {
+                sendReport(reportId.get(), reporter, false);
+            }
         }
-
         return result;
     }
+
+    private void sendReport(UUID networkUuid, Reporter reporter, boolean overwrite) {
+        var restTemplate = new RestTemplate();
+        var headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        var resourceUrl = reportServerURI + DELIMITER + REPORT_API_VERSION + DELIMITER + "report" + DELIMITER + networkUuid.toString();
+        var uriBuilder = UriComponentsBuilder.fromHttpUrl(resourceUrl).queryParam("overwrite", overwrite);
+        try {
+            restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.PUT, new HttpEntity<>(objectMapper.writeValueAsString(reporter), headers), ReporterModel.class);
+        } catch (Exception error) {
+            LOGGER.error(error.getMessage());
+        }
```

Comment:
throw new PowsyblException(error); ?
To have a feedback to the front

Reply:
done